### PR TITLE
Take advantage of some new functionality and features from Kotlin 1.3

### PR DESCRIPTION
**Pull request by SackCastellon**
This PR updates some code to take advantage of some new Koltin 1.3 features like:
- Contracts`, allowing for proper smart-cast when using:
  - `isNullOrEmpty()`
  - `isNullOrBlank()`
  - `check(Boolean)`
  - `require(Boolean)`
- `hashCode()` with nullable receiver
- Capturing `when` subject in a variable

### CODE REVIEW

File: src/main/java/tornadofx/LayoutDebugger.kt [hunk 1]
```diff
@@ -406,7 +406,7 @@ class LayoutDebugger : Fragment() {
         override fun toString(v: Insets?) = if (v == null) "" else "${v.top.s()} ${v.right.s()} ${v.bottom.s()} ${v.left.s()}"
         override fun fromString(s: String?): Insets {
             try {
-                if (s == null || s.isEmpty()) return Insets.EMPTY
+                if (s.isNullOrBlank()) return Insets.EMPTY
```
Comment (ruckustboom) [inline src/main/java/tornadofx/LayoutDebugger.kt]: This is a semantic change. Did you mean `s.isNullOrEmpty()`?

File: src/main/java/tornadofx/Rest.kt [hunk 2]
```diff
@@ -166,12 +166,10 @@ open class Rest : Controller() {
             try {
                 val content = text()
 
-                if (content == null || content.isEmpty())
+                if (content.isNullOrBlank())
```
Comment (ruckustboom) [inline src/main/java/tornadofx/Rest.kt]: This is a semantic change. Did you mean `isNullOrEmpty()`?

File: src/main/java/tornadofx/LayoutDebugger.kt (hunk 1 above)
Comment (ruckustboom) [inline src/main/java/tornadofx/LayoutDebugger.kt]: This is a semantic change. Did you mean `isNullOrEmpty()`?

File: src/main/java/tornadofx/Rest.kt (hunk 2 above)
Comment (ruckustboom) [inline src/main/java/tornadofx/Rest.kt]: This is a semantic change. Did you mean `isNullOrEmpty()`?

File: src/main/java/tornadofx/Rest.kt [hunk 3]
```diff
@@ -187,12 +185,10 @@ open class Rest : Controller() {
...
```
Comment (ruckustboom) [inline src/main/java/tornadofx/Rest.kt]: This is a semantic change. Did you mean `isNullOrEmpty()`?

**Comment by ruckustboom** [comment]
I tried to comment in the code, but GitHub is glitching for me pretty badly, so I'm not sure what you are seeing.

There are a number of places in this PR where `thing == null || thing.isEmpty()` was replaced with `this.isNullOrBlank()`, which is a semantic change.

File: src/main/java/tornadofx/LayoutDebugger.kt (hunk 1 above)
Comment (SackCastellon) [reply src/main/java/tornadofx/LayoutDebugger.kt]: You are right it is a semantic change, but in this case, whether the string is empty or blank the result will be `Insets.EMPTY`, in the old code if the string was blank an exception was thrown and then ignored to return `Insets.EMPTY`. In the new code we prevent the exception to be thrown as we know that if it is black we will return `Insets.EMPTY`.

In my opinion this change is justified.

File: src/main/java/tornadofx/Rest.kt (hunk 2 above)
Comment (SackCastellon) [reply src/main/java/tornadofx/Rest.kt]: In this case, you are right, I think the change is not justified. I will change it to `isNullOrEmpty()`

File: src/main/java/tornadofx/Rest.kt (hunk 3 above)
Comment (SackCastellon) [reply src/main/java/tornadofx/Rest.kt]: In this case, you are right, I think the change is not justified. I will change it to `isNullOrEmpty()`

**Comment by edvin** [comment]
Thanks guys, good work! Nice to start utilizing the new features of Kotlin 1.3 :)

**Comment by ruckustboom** [comment]
@SackCastellon I agree `isNullOrBlank()` would make more sense in a lot of those places, it's just that it is a semantic change out of scope of the rest of this PR ("Take advantage of some new functionality and features from Kotlin 1.3"). I would recommend creating a second PR with those changes to separate logical corrections from syntactic updates.

**Pull request merged**